### PR TITLE
Fix crash in -Xsource:3 varargs

### LINKED ISSUE #12415: Unexpected tree in -Xsource:3 vararg splice
## reproduction steps

When using `-Xsource:3` with the following code:

```scala
object Test {
  val a = Seq(1, 2)
  println(Seq((a ++ a)*))
}
```

## problem

The compiler fails with:
```
scalac: 
  unexpected tree: class scala.tools.nsc.ast.Trees$Parens
Parens(List(a.$plus$plus(a)))
     while compiling: /home/tobi/workspace/test-projects/plain-scala-proj/src/main/scala/Test.scala
        during phase: typer
     library version: version 2.13.6
    compiler version: version 2.13.6
  reconstructed args: -classpath /home/tobi/workspace/test-projects/plain-scala-proj/target/scala-2.13/classes:/home/tobi/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.6/scala-library-2.13.6.jar:/home/tobi/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.13.6/scala-reflect-2.13.6.jar:/home/tobi/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalactic/scalactic_2.13/3.2.5/scalactic_2.13-3.2.5.jar:/home/tobi/workspace/test-projects/plain-scala-proj/lib/kotlin-reflect-sources.jar:/home/tobi/workspace/test-projects/plain-scala-proj/lib/kotlin-reflect.jar:/home/tobi/workspace/test-projects/plain-scala-proj/lib/kotlin-stdlib-jdk7-sources.jar:/home/tobi/workspace/test-projects/plain-scala-proj/lib/kotlin-stdlib-jdk7.jar:/home/tobi/workspace/test-projects/plain-scala-proj/lib/kotlin-stdlib-jdk8-sources.jar:/home/tobi/workspace/test-projects/plain-scala-proj/lib/kotlin-stdlib-jdk8.jar:/home/tobi/workspace/test-projects/plain-scala-proj/lib/kotlin-stdlib-sources.jar:/home/tobi/workspace/test-projects/plain-scala-proj/lib/kotlin-stdlib.jar:/home/tobi/workspace/test-projects/plain-scala-proj/lib/kotlin-test-sources.jar:/home/tobi/workspace/test-projects/plain-scala-proj/lib/kotlin-test.jar -Xsource:3.0.0

  last tree to typer: Parens
       tree position: line 5 of /home/tobi/workspace/test-projects/plain-scala-proj/src/main/scala/Test.scala
              symbol: null
           call site: object Test in package <empty>

== Source file context for tree position ==

     2 
     3 object Test {
     4   val a = Seq(1, 2)
     5   println(Seq((a ++ a)*))
     6 }
     7 
```

### PR DESCRIPTION
**Pull request by som-snytt**
Fix crash in -Xsource:3 varargs.

Fixes scala/bug#12415

### CODE REVIEW

File: src/compiler/scala/tools/nsc/ast/parser/Parsers.scala [hunk 1]
```diff
@@ -2171,7 +2171,7 @@ self =>
         }
         def checkWildStar: Tree = top match {
           case Ident(nme.WILDCARD) if isSequenceOK && isRawStar => peekingAhead (
-            if (isCloseDelim) atPos(top.pos.start, in.prev.offset)(Star(stripParens(top)))
+            if (isCloseDelim) atPos(top.pos.start, in.prev.offset)(Star(top))
```
Comment (smarter) [inline src/compiler/scala/tools/nsc/ast/parser/Parsers.scala]: Why was stripParens removed here?

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/ast/parser/Parsers.scala]: law of conservation of stripParens. Also, top was just matched on Ident so no parens. Then where did the parens go?

I'm about to push another commit with minor refactor.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/ast/parser/Parsers.scala]: Oh right, I think I saw this debated on dotty:
```
scala> List(42) match { case Seq((xs)*) => xs }
1 |List(42) match { case Seq((xs)*) => xs }
  |                               ^
  |                               `*` must follow pattern variable
```
even though parens look innocent here.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/ast/parser/Parsers.scala]: Scala 2 is ok with
```
List(42) match { case Seq((xs) @ _*) => xs }
```
but not
```
scala> List(42) match { case Seq((xs)*) => xs }
                                      ^
       error: bad simple pattern: use _* to match a sequence
```
under `-Xsource:3`

**Review by lrytz — APPROVED** [review]

**Comment by lrytz** [comment]
Thank you!

**Pull request merged**